src/librustc_middle/mir/interpret/error.rs

```diff
@@ -305,7 +305,7 @@ impl fmt::Debug for InvalidProgramInfo<'_> {
             Layout(ref err) => write!(f, "{}", err),
             TransmuteSizeDiff(from_ty, to_ty) => write!(
                 f,
-                "tried to transmute from {:?} to {:?}, but their sizes differed",
+                "transmuting from {:?} to {:?}, which do not have the same size",
```

Comment:
while we're bikeshedding diagnostic messages:

```suggestion
                "transmuting `{}` to `{}` is not possible, because these types do not have the same size",
```